rcl_lifecycle/src/default_state_machine.c

```diff
@@ -695,14 +699,27 @@ rcl_lifecycle_init_default_state_machine(
 fail:
   // If rcl_lifecycle_transition_map_fini() fails, it will clobber the error string here.
   // Concatenate the error strings if that happens
+  default_allocator = rcl_get_default_allocator();
+
   if (rcl_error_is_set()) {
-    fail_error_message = rcl_get_error_string().str;
+    init_error_string = rcutils_strdup(rcl_get_error_string().str, default_allocator);
+    if (init_error_string != NULL) {
+      fail_error_message = init_error_string;
+    } else {
+      fail_error_message = "Error while duplicating strings by rcutils_strdup() !";
+    }
+    rcl_reset_error();
   }
 
   if (rcl_lifecycle_transition_map_fini(&state_machine->transition_map, allocator) != RCL_RET_OK) {
     const char * fini_error = "";
     if (rcl_error_is_set()) {
-      fini_error = rcl_get_error_string().str;
+      fini_error_string = rcutils_strdup(rcl_get_error_string().str, default_allocator);
```

Comment:
So I think this should not be a `strdup` but rather a `snprintf` concatenating two error messages.

Reply:
The below code has concatenating two error messages. So not do this here.  

https://github.com/ros2/rcl/blob/bd6fbb3d45c271f7351c9d0932b0c7072bd57b1c/rcl_lifecycle/src/default_state_machine.c#L708-L711